src/twisted/runner/test/test_procmon.py

```diff
@@ -118,6 +121,8 @@ def spawnProcess(
         Fake L{reactor.spawnProcess}, that logs all the process
         arguments and returns a L{DummyProcess}.
         """
+        if self._spawn_resource_unavailable:
+            raise BlockingIOError("[Errno 35] Resource temporarily unavailable")
```

Comment:
You're using the constructor for `BlockingIOError` incorrectly here.  What you want is `BlockingIOError(errno.EAGAIN, os.strerror(EAGAIN))` .  You should use the errno so nobody needs "35" memorized.

But also… `EAGAIN` *in particular* seems like a weird error for `spawnProcess` not to handle itself. It seems like a bug; in other async operations it's Twisted's job to handle EWOULDBLOCK / EAGAIN (etc) for you so you don't need to think about it. When and why does this happen?